extension/src/test/suite/experiments/data/index.test.ts

```diff
@@ -67,22 +76,69 @@ suite('Experiments Data Test Suite', () => {
       )
     })
 
-    it('should watch the .git directory for updates', async () => {
+    it('should watch the .git directory for updates when the directory is outside of the workspace', async () => {
       const mockNow = getMockNow()
       const gitRoot = dvcDemoPath
 
-      const mockExecuteCommand = (command: CommandId) => {
-        if (command === AvailableCommands.GIT_GET_REPOSITORY_ROOT) {
-          return Promise.resolve(gitRoot)
-        }
-      }
+      const data = disposable.track(
+        new ExperimentsData(
+          dvcDemoPath,
+          {
+            dispose: stub(),
+            executeCommand: (command: CommandId) => {
+              if (command === AvailableCommands.GIT_GET_REPOSITORY_ROOT) {
+                return Promise.resolve(gitRoot)
+              }
+            }
+          } as unknown as InternalCommands,
+          disposable.track(new EventEmitter<boolean>())
+        )
+      )
+
+      await data.isReady()
+      bypassProcessManagerDebounce(mockNow)
+
+      const managedUpdateSpy = spy(data, 'managedUpdate')
+      const dataUpdatedEvent = new Promise(resolve =>
+        data.onDidUpdate(() => resolve(undefined))
+      )
+
+      await Watcher.fireWatcher(
+        FileSystem.getGitPath(gitRoot, gitPath.DOT_GIT_HEAD)
+      )
+      await dataUpdatedEvent
+
+      expect(managedUpdateSpy).to.be.called
+    })
+
+    it('should watch the .git directory for updates when the directory is inside workspace', async () => {
+      const mockNow = getMockNow()
+      const gitRoot = dvcDemoPath
+      const mockDotGitFilePath = join(
+        MOCK_WORKSPACE_GIT_FOLDER,
+        gitPath.DOT_GIT_HEAD
+      )
+      const mockDotGitNestedFilePath = join(
+        MOCK_WORKSPACE_GIT_FOLDER,
+        EXPERIMENTS_GIT_LOGS_REFS,
+        'index'
+      )
+
+      ensureFileSync(mockDotGitFilePath)
+      ensureFileSync(mockDotGitNestedFilePath)
+
+      stub(FileSystem, 'getGitPath').returns(MOCK_WORKSPACE_GIT_FOLDER)
 
       const data = disposable.track(
         new ExperimentsData(
           dvcDemoPath,
           {
             dispose: stub(),
-            executeCommand: mockExecuteCommand
+            executeCommand: (command: CommandId) => {
```

Comment:
Ended up moving the `executeCommand` to be inline to avoid the `no-identical-function` eslint rule. I could be wrong, but when I tried to create a single function for the two to share it looked like I was overcomplicating things 🤔